axelor-stock/src/main/java/com/axelor/apps/stock/web/LogisticalFormController.java

```diff
@@ -126,4 +130,26 @@ public void setCustomerAccountNumberToCarrier(ActionRequest request, ActionRespo
       TraceBackService.trace(response, e);
     }
   }
+
+  public void refreshProductNetMass(ActionRequest request, ActionResponse response) {
+    try {
+      LogisticalForm logisticalForm = request.getContext().asType(LogisticalForm.class);
+
+      for (LogisticalFormLine logisticalFormLine : logisticalForm.getLogisticalFormLineList()) {
+        if (logisticalFormLine.getTypeSelect() == LogisticalFormLineRepository.TYPE_DETAIL) {
+          Long id = logisticalFormLine.getStockMoveLine().getProduct().getId();
+          Product product =
+              Beans.get(ProductRepository.class).all().filter("self.id = ?", id).fetchOne();
+          if (product == logisticalFormLine.getStockMoveLine().getProduct()) {
+            logisticalFormLine.setUnitNetMass(product.getNetMass());
+
+            response.setValue("logisticalFormLineList", logisticalForm.getLogisticalFormLineList());
+          }
```

Comment:
Keep **response.setValue("logisticalFormLineList", logisticalForm.getLogisticalFormLineList());** out of for loop